nasa-gibs/worldview
# WV-2617 Ability to change palette of single-color vector products

## Description
It's currently possible to change the palette of our single-color raster products ([MODIS Sea Ice example](https://worldview.earthdata.nasa.gov/?v=-144.07583239055444,21.320852988221986,-45.99786689931894,89.45841674019755&l=Reference_Labels_15m(hidden),Reference_Features_15m(hidden),Coastlines_15m,MODIS_Terra_Sea_Ice,MODIS_Aqua_CorrectedReflectance_TrueColor(hidden),MODIS_Terra_CorrectedReflectance_TrueColor(hidden),Land_Water_Map&lg=true&t=2023-02-13-T02%3A00%3A00Z)) but not possible to change our single-color vector products ([fires, nuclear, etc. example)](https://worldview.earthdata.nasa.gov/?v=38.574142522617386,21.95807570900159,60.78583562877069,39.122934748630186&l=Coastlines_15m,GRanD_Reservoirs,Nuclear_Power_Plant_Locations,MODIS_Combined_Thermal_Anomalies_All,VIIRS_NOAA20_Thermal_Anomalies_375m_All,MODIS_Terra_CorrectedReflectance_TrueColor&lg=true&t=2022-09-16-T00%3A00%3A00Z).

We should add this ability to our single-color vector products, excluding Orbit tracks & Settlements which pose problems due to the nature of their WMS tiles.

- Fixes # wv-2617

## How To Test

- git checkout wv-2617-change-palette
- npm ci
- npm run watch
- Load any vector product (i.e. [Reservoirs](http://localhost:3000/?v=-112.57070732457854,12.791196095421473,-36.416658428611804,68.61412952088847&l=GRanD_Reservoirs,Coastlines_15m,MODIS_Terra_CorrectedReflectance_TrueColor&lg=false&t=2023-05-17-T19%3A58%3A42Z))
- Click the layer info button & notice the Color Palette component is present.
- Select any color from the palette 

Vector layers are treated differently depending on the zoom level, so we need to confirm the palette is applied in both cases.
- **When zoomed out**. If you click on a vector feature while **zoomed out** you will get a "Vector features may not be clickable at all zoom levels" message.
- **When zoomed in**. Clicking on a vector feature while  **zoomed in** provides a window of information about the feature. 

@nasa-gibs/worldview
